### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by khalilhamdii**
I added the following features in this branch : 

- Changed App into a class component.
- Initialized the App component with three properties.
- Implemented the handleClick function which takes buttonName as an argument and changes the state based on the result of the calculate function.
- Passed the total or next value to Display to display the current result.
- Passed handleClick as a clickHandler property to ButtonPanel
- Implemented the handleClick function which takes buttonName as an argument and returns clickHandler from props.
- Passed the clickHandler property to each button in the panel.
- Implemented the handleClick function which takes buttonName as an argument and returns clickHandler from props.
- Used the onClick event to attach the handleClick event to the button.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+      next: modifiedData.next,
+      operation: modifiedData.operation,
+    });
+  }
+
+  render() {
+    const { total } = this.state;
+    return (
+      <>
+        <Display result={total} />
```
Comment (bellom) [inline src/components/App.js:34]: ### [Optional]

You should use OR operator in here...
your calculation is always starting with `0` when a user clicks `a/c` then click a number(5), we have `05`.

example: `<Display result={next || total || '0'} />`

File: src/logic/calculate.js [hunk 2]
```diff
@@ -6,27 +6,24 @@ const calculate = (data, btnName) => {
...
-      total += btnName;
+      total = total === null ? btnName : total + btnName;
     } else if (total && operation) {
-      next += btnName;
+      next = next === null ? btnName : next + btnName;
     }
   } else if (btnName === 'AC') {
     total = '0';
     next = '';
     operation = '';
```
Comment (bellom) [inline src/logic/calculate.js:16]: ### [Optional]
Your calculation is always starting with `0` when a user clicks `a/c` then click a number(5), we have `05`.

You might need to set all(total, next, operation) to `null`

**Review by bellom — APPROVED** [review]
## Hi @khalilhamdii  :clap: 

##### You have done wonderfully well, you passed all checklist for this milestone :+1:
 
**Update your codebase before merging, checkout comment below, I trust you can do this.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion 

**Best,**

**Oluwaseun J. Bello**
**_Slack Handle - @sean bellom_**

File: src/logic/calculate.js (hunk 2 above)
Comment (khalilhamdii) [reply src/logic/calculate.js]: Thank you for your review! I fix the AC btn to set them all to null

**Pull request merged**